dateutil/parser.py

```diff
@@ -602,8 +602,6 @@ def parse(self, timestr, default=None, ignoretz=False, tzinfos=None, **kwargs):
                     raise ValueError("Offset must be tzinfo subclass, "
                                      "tz string, or int offset.")
                 ret = ret.replace(tzinfo=tzinfo)
-            elif res.tzname and res.tzname in time.tzname:
```

Comment:
After thinking more about the problem, I agree with you.

I was originally thinking about the problem for scenarios where a date string is produced by a machine - in such cases, we can probably expect that a timezone in a date string is significant and should be honored. For example, the date string `'2016-10-10T08:00:00 CST'` means `2016-10-10 08:00:00-06:00` because `CST` is `UTC-06:00`.

However, in many cases the source of a date string may be from user input where the user considers the time to be authoritative and the timezone to be informative but not necessarily correct. I would call this a "colloquial" date string. For example, a user may write `October 10, 8am CST`, expecting the time to be `2016-10-10T08:00:00-05:00` because CST was given to mean US Central time, not exactly Central Standard Time.

My previous proposal would violate the expectations for colloquial date strings because `CST` would be taken explicitly.

I'm now wondering if the behavior of dateutil actually needs to change for this. Similarly to your suggestion on #318, the `parse` caller could generally enforce the offset of local timezones by passing `tzinfos={time.tzname[0]: -time.timezone, time.tzname[1]: -time.altzone}`. Having these examples in the documentation would probably be sufficient to avoid issues like the one that was observed by botocore users in the UK.